src/promptflow/promptflow/_sdk/schemas/_flow.py

```diff
@@ -58,6 +63,22 @@ class EagerFlowSchema(BaseFlowSchema):
     """Schema for eager flow."""
 
     # path to flow entry file.
-    path = LocalPathField(required=True)
+    path = LocalPathField(required=False)
     # entry function
     entry = fields.Str(required=True)
+
+    @post_load
+    def infer_path(self, data: dict, **kwargs):
```

Comment:
Why not use `@validates_schema` as offline synced?
https://marshmallow.readthedocs.io/en/latest/extending.html#schema-level-validation

Reply:
here the major purpose is to infer path before your PR merged. Will update it after your PR merged.